### ISSUE TITLE
Design user profile editor and viewer

### ISSUE THREAD

**Issue by hannahkane** [mozilla/teach.webmaker.org#711]
A main purpose of the Teach site is to connect our community members to one another, so an important feature will be to allow users to create rich profiles that showcase their skills and interests. Later iterations will focus on search and discovery among the profiles.

### Fields to include: ###

User entered data 
 * name
 * photo
 * location
 * contact info 
 * languages
 * skills
 * interests

Tell us how and where you teach. *
We know you teach others in a variety of ways. Please select all that apply.

    At standalone events (for example, a one-day workshop, hackathon or Maker Party event)
    In a classroom during the school day
    As part of an after-school or summer program
    With my friends and family at home
    At professional meet-ups or training events with other adults/mentors/educators
    At a college or university
    At a library or other community space
    Other: 

Approximately how many learners do you reach in a year? *
Please select one option.

    0-50
    51-100
    101-200
    201-500
    More than 500 

What is the age range of your learners? *

    6-13
    14-24
    25-34
    35-44
    45-54
    55+ 

Tell us briefly about your motivations. Why are you interested in teaching others how to read, write and participate in the digital world.
(300 characters max)

System data
 * join date
 * groups/listings owned
 * badges earned

We should design both an editor and a viewer. We should consider scenarios where the user has entered a lot of information, as well as scenarios where they've left many fields blank.

**Comment by hannahkane** [comment]
Question for @sabrinang and @iamjessklein - should I create a separate ticket for the UX pieces of this, or does it make sense to work together on a shared ticket?

UX stuff I can think of:
 * How does the user access their profile - to create it or edit it?
 * Process for saving their profile
 * Process for deleting
 * Photo upload UX
 * Location UX (same as on the Clubs form?)
 * Languages, Skills, Interests UX (defined lists? tags?)

**Comment by davidascher** [comment]
If we think a high percentage of users will already have Mozillians.org
profiles, we may want to explore a "get data from mozillians.org" mode and
"linking" the accounts.  May be too complicated, but should be assessed.

**Comment by hannahkane** [comment]
#585 is likely a pre-req

**Comment by hannahkane** [comment]
To consider:
* Privacy preferences / defaults (consider a geography-based setting)
* Reputation
* Letting people tag other people's profiles

**Comment by hannahkane** [comment]
This should happen before the nav stuff in #979 because it will inform which API endpoints we build.

**Comment by hannahkane** [comment]
This is duped by latest work
